Tools/SetupUnityPackage/UnityUtils/FileHelper.cs

```diff
@@ -33,16 +33,22 @@ public static string GetInternalStorage()
 
             // On Android persistentDataPath returns the external folder, where the app may not have permissions to write.
             // we use reflection to call File.getAbsolutePath(currentActivity.getFilesDir())
-            using (var unityPlayer = new AndroidJavaClass("com.unity3d.player.UnityPlayer"))
-            using (var currentActivity = unityPlayer.GetStatic<AndroidJavaObject>("currentActivity"))
-            {
-                var getFilesDir = AndroidJNIHelper.GetMethodID(AndroidJNI.FindClass("android/content/ContextWrapper"), "getFilesDir", "()Ljava/io/File;");
-                var internalFilesDir = AndroidJNI.CallObjectMethod(currentActivity.GetRawObject(), getFilesDir, Array.Empty<jvalue>());
-                var getAbsolutePath = AndroidJNIHelper.GetMethodID(AndroidJNI.FindClass("java/io/File"), "getAbsolutePath", "()Ljava/lang/String;");
+            var path = string.Empty;
 
-                var path = AndroidJNI.CallStringMethod(internalFilesDir, getAbsolutePath, Array.Empty<jvalue>());
-                return path ?? $"/data/data/{Application.identifier}/files";
+            try
+            {
+                using (var unityPlayer = new AndroidJavaClass("com.unity3d.player.UnityPlayer"))
+                using (var currentActivity = unityPlayer.GetStatic<AndroidJavaObject>("currentActivity"))
+                {
+                    var filesDir = currentActivity.Call<AndroidJavaObject>("getFilesDir");
+                    path = filesDir.Call<string>("getAbsolutePath");
```

Comment:
You can return here - no need to declare `path` at all.

Reply:
This was just set that way just because of the try/catch. It'll change since, as you noticed, the consumer handles exceptions.

Reply:
For future reference, even in a try-catch, you can do a return:

```csharp
try
{
    return DoSomethingThatMayThrow();
}
catch
{
}

return null;
```

Reply:
That makes sense since the return type is a reference type.